### PR TITLE
Streamline regex

### PR DESCRIPTION
**Pull request by KayleighB94**

### CODE REVIEW

File: wrapper_writer/parsers.py [hunk 1]
```diff
@@ -76,123 +77,84 @@ def delete_config(self):
...
-        """
-        matches = re.finditer(r"/\*\*([\s,\w,\*,@,\+\.,='\[\]\-/%]*)\*/", data, re.MULTILINE)
+        matches = re.finditer(self.regex_string, data, re.MULTILINE)
+        methods = []
+        doc_string = ""
         for match in matches:
-            group = match.group(1)
-            if group:
-                doc = group.replace("*", "").replace("\n     @", "\n@").replace("\n    ", "").replace("\n     ", " ")
+            if match.group(2) is None:
```
Comment (treilly94) [inline wrapper_writer/parsers.py]: Comments through here might be useful, its difficult to tell whats going on

File: wrapper_writer/parsers.py [hunk 2]
```diff
@@ -76,123 +77,84 @@ def delete_config(self):
             print("The config file does not exists")
 
 
-class ScalaParse(Parser):
+class ScalaParser(Parser):
     """
     This ScalaParse class parses a scala file, extracts the method elements and writes them out to a config file
```
Comment (treilly94) [inline wrapper_writer/parsers.py]: This ScalaParser

File: wrapper_writer/parsers.py [hunk 3]
```diff
@@ -76,123 +77,84 @@ def delete_config(self):
...
-            self.doc_strings.append(doc.strip())
-
-    @staticmethod
-    def extract_return_type(raw_res):
+    def create_containers(self, methods, file_path):
         """
-        This function extracts the return type of a method
-        :param raw_res: A method signature from the file to be parsed
-        :return: return_type: String object of the return type
+        This function
```
Comment (treilly94) [inline wrapper_writer/parsers.py]: this function?

File: wrapper_writer/parsers.py (hunk 2 above)
Comment (KayleighB94) [reply wrapper_writer/parsers.py]: ?? what about it

Comment (KayleighB94) [reply wrapper_writer/parsers.py]: done

File: wrapper_writer/parsers.py (hunk 1 above)
Comment (KayleighB94) [reply wrapper_writer/parsers.py]: done

File: wrapper_writer/parsers.py (hunk 3 above)
Comment (KayleighB94) [reply wrapper_writer/parsers.py]: hehe

Comment (KayleighB94) [reply wrapper_writer/parsers.py]: done

**Pull request merged**